src/ios/FirebasePerformancePlugin.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  FirebasePerformancePlugin.swift
+//  Firebase Sample App
+//
+//  Created by Carlos Correa on 12/03/2021.
+//
+
+import Foundation
+import Firebase
+
+class FirebasePerformancePlugin {
+    var traces: [String: Trace] = [:]
+    
+    func starTrace(traceName: String){
+        if !traceName.isEmpty {
+            if let trace = traces[traceName] {
+                trace.start()
+            } else {
+                let trace = Performance.startTrace(name: traceName)
+                traces[traceName] = trace
+            }
+        }
+    }
+    
+    func stopTrace(traceName: String) {
+        if !traceName.isEmpty {
+            if let trace = traces[traceName] {
+                trace.stop()
+            } else {
+                let trace = traces[traceName]
+                trace?.stop()
+            }
+        }
+    }
+
+    func addTraceAttribute(traceName: String, attributeName: String ,value: String) {
+        if !traceName.isEmpty && !attributeName.isEmpty && !value.isEmpty {
+            if let trace = traces[traceName] {
```

Comment:
Same as above this can be easier to read :)  :

```
guard 
         !traceName.isEmpty, 
         !attributeName.isEmpty, 
         !value.isEmpty, 
         let trace = traces[traceName] 
else { return }

trace.setValue(attributeName, forAttribute: value)

```